Tasks/DotNetCoreInstallerV1/versioninstaller.ts

```diff
@@ -0,0 +1,167 @@
+            }
+
+            // Cache tool
+            this.createInstallationCompleteFile(versionInfo);
+
+            console.log(tl.loc("SuccessfullyInstalled", this.packageType, version));
+        }
+        catch (ex) {
+            throw tl.loc("FailedWhileInstallingVersionAtPath", version, this.installationPath, JSON.stringify(ex));
+        }
+        finally {
+            //todo: Release Lock and stop timer
+        }
+
+        //todo: in case of failure due to: unable to acquire lock or timeout, 3 retries to install.
+    }
+
+    public isVersionInstalled(version: string): boolean {
+        if (!toolLib.isExplicitVersion(version)) {
+            throw tl.loc("ExplicitVersionRequired", version);
+        }
+
+        var isInstalled: boolean = false;
+        if (this.packageType == utils.Constants.sdk) {
+            isInstalled = tl.exist(path.join(this.installationPath, utils.Constants.relativeSdkPath, version)) && tl.exist(path.join(this.installationPath, utils.Constants.relativeSdkPath, `${version}.complete`));
+        }
+        else {
+            isInstalled = tl.exist(path.join(this.installationPath, utils.Constants.relativeRuntimePath, version)) && tl.exist(path.join(this.installationPath, utils.Constants.relativeRuntimePath, `${version}.complete`));
+        }
+
+        isInstalled ? console.log(tl.loc("VersionFoundInCache", version)) : console.log(tl.loc("VersionNotFoundInCache", version));
+        return isInstalled;
+    }
+
+    private createInstallationCompleteFile(versionInfo: VersionInfo): void {
+        tl.debug(tl.loc("CreatingInstallationCompeleteFile", versionInfo.version, this.packageType));
+        // always add for runtime as it is installed with sdk as well.
+        var pathToVersionCompleteFile: string = "";
+        if (this.packageType == utils.Constants.sdk) {
+            let sdkVersion = versionInfo.version;
+            pathToVersionCompleteFile = path.join(this.installationPath, utils.Constants.relativeSdkPath, `${sdkVersion}.complete`);
+            tl.writeFile(pathToVersionCompleteFile, `{ "version": "${sdkVersion}" }`);
+        }
+
+        let runtimeVersion = VersionInfo.getRuntimeVersion(versionInfo, this.packageType);
+        if (runtimeVersion) {
+            pathToVersionCompleteFile = path.join(this.installationPath, utils.Constants.relativeRuntimePath, `${runtimeVersion}.complete`);
+            tl.writeFile(pathToVersionCompleteFile, `{ "version": "${runtimeVersion}" }`);
+        }
+        else if (this.packageType == "runtime") {
+            throw tl.loc("CannotFindRuntimeVersionForCompletingInstallation", this.packageType, versionInfo.version);
+        }
+    }
+
+    private isLatestInstalledVersion(version: string): boolean {
+        var pathTobeChecked = this.packageType == utils.Constants.sdk ? path.join(this.installationPath, utils.Constants.relativeSdkPath) : path.join(this.installationPath, utils.Constants.relativeRuntimePath);
```

Comment:
lets copy the child items only when the sdk is installed

Reply:
Will this have some good advantage ?